app/src/main/kotlin/at/bitfire/davdroid/repository/DavCollectionRepository.kt

```diff
@@ -288,4 +347,34 @@ class DavCollectionRepository @Inject constructor(
     private fun getVTimeZone(tzId: String): String? =
         DateUtils.ical4jTimeZone(tzId)?.toString()
 
-}
\ No newline at end of file
+
+    /*** OBSERVERS ***/
+
+    fun addOnChangeListener(observer: OnChangeListener) = synchronized(observers) {
```

Comment:
If we use a thread-safe collection, we don't need to synchronize here. I suggest `observers =mutableSetOf(…)`.

Reply:
How is `observers = mutableSetOf()` (more) thread-safe ? 

I have changed it anyhow. But would be nice with a short explanation :)

Reply:
Seems that I have remembered that incorrectly. `mutableSetOf` is not thread-safe by default. So we can use a thread-safe `Set` implementation or wrap it with `Collections.synchronizedSet`.